### ISSUE TITLE
Does twitter_auth.sh still work?

### ISSUE THREAD

**Issue by dhdurgee** [cmj/nitter#1]
I successfully built Nitter with the cookie_header branch via these instructions:

[https://github.com/cmj/nitter/wiki/Install](https://github.com/zedeus/nitter/issues/url)

I am now at the stage where I am attempting to modify the basic nitter.conf and am getting the following with the script:

`dhdurgee@A31524PT:/opt/nitter$ ./twitter_auth.sh

cookieHeader = "ct0=; auth_token="
xCsrfToken = ""
dhdurgee@A31524PT:/opt/nitter$ `

Does this script still work?

**Comment by cmj** [comment]
I should preface this by saying you are given 2 options in the wiki on how to obtain auth_token and csrf...

That said I just checked and the script still seems to work for me. It takes at least 1 try on my end.

Your output suggests some tool is missing in the script, do you have `jq` installed?

This should be the output:
> ![](https://camo.githubusercontent.com/ae1e25af4ffc08290c918806755cf7702c6cce8533c83c56a04b4635ed91d563/68747470733a2f2f692e696d6775722e636f6d2f5562317a6269412e706e67)

**Comment by cmj** [comment]
This is the latest attempt. I'm not sure where it's failing on your end.

https://gist.github.com/cmj/17fa133a948eedd0167bdcbff1dfff19

**Comment by dhdurgee** [comment]
Here is what I get:

`dhdurgee@A31524PT:/opt/nitter$ ./twitter-cookie.sh 
--- nitter.conf ---
cookieHeader = "ct0=; auth_token="
xCsrfToken = ""
--- cookies.json ---
{"att":"1-tnK8ZVyTqxh8Zd0F6FeMn6fTvXpl5o8fN2oYGkOe","guest_id_marketing":"v1%3A173134753462989335","guest_id_ads":"v1%3A173134753462989335","personalization_id":"\"v1_k+JMQ3E6WXzqa/X6kjX6Rg==\"","guest_id":"v1%3A173134753462989335"}
dhdurgee@A31524PT:/opt/nitter$ 

`

**Comment by cmj** [comment]
So just to be clear, you are getting the `cookieHeader` and `xCsrfToken` values, you just removed them from the comment (for good reason)?

If so, enter those two lines into nitter.conf:
https://github.com/cmj/nitter/blob/d89893f1769327be42323c077a9d148d6b9c0c5a/nitter.example.conf#L38-L39

**Comment by dhdurgee** [comment]
No, I simply copy/pasted from my terminal window.  I thought seeing exactly what is being returned would be of assistance to you in determining what is failing.  Per you earlier question:

`dhdurgee@A31524PT:/opt/nitter$ echo $SHELL
/bin/bash
dhdurgee@A31524PT:/opt/nitter$ 


and 

dhdurgee@A31524PT:/opt/nitter$ which jq
/usr/bin/jq
dhdurgee@A31524PT:/opt/nitter$

**Comment by cmj** [comment]
There has been some suspicion that Twitter is flagging/denying known commercial hosting ip ranges too. Running on my residential network I'm not seeing issues, but if i run it on some public shell servers, the login flow fails at username every single time. So this might be the overall issue.

Testing on numerous accounts:

```
# Line 47: jq -r '.subtasks[0].cta.secondary_text.text' <<< "${flow_3}"; exit

 ~$ ./twitter_auth.sh
We blocked an attempt to access your account because we weren't sure it was really you.

This happens when we notice unusual login activity, like an attempt to log in too many times, or from a different location or device.
```

**Comment by dhdurgee** [comment]
I am not using a VPN, I am on a residential internet connection with Century Link.  The IP address is: 174.29.65.18 here.

**Comment by cmj** [comment]
All right, I'm going to clean a few things up in the next couple days.  Thanks for your feedback. I ping you with an update.

**Comment by alphaleadership** [comment]
for windows this dont work

**Comment by alphaleadership** [comment]
jq dont exist

**Comment by cmj** [comment]
@dhdurgee I updated some elements to the script. I have it loop 3 times as well. Try this when you get a chance:
https://gist.github.com/cmj/408a3c003834526c97804e8fa5829afe

Some testing shows I can auth successfully once every 60 seconds. One account kept authing properly almost ever 5 seconds, so there are some things I don't understand at play on their end. 

![](https://i.imgur.com/0hmwZpS.png)

@alphaleadership I don't have Windows handy at the moment... also https://jqlang.github.io/jq/download/

**Comment by dhdurgee** [comment]
I regret to say that I am seeing no change here as you can see:

`dhdurgee@A31524PT:/opt/nitter$ ./twitter_auth.sh 
cookieHeader = "ct0=; auth_token="
xCsrfToken = ""
dhdurgee@A31524PT:/opt/nitter$ ./twitter_auth.sh 
cookieHeader = "ct0=; auth_token="
xCsrfToken = ""
dhdurgee@A31524PT:/opt/nitter$ ./twitter_auth.sh 
Suspicious login prevented
Retry 1/3 in 20 seconds...
cookieHeader = "ct0=; auth_token="
xCsrfToken = ""
dhdurgee@A31524PT:/opt/nitter$ ./twitter_auth.sh 
cookieHeader = "ct0=; auth_token="
xCsrfToken = ""
dhdurgee@A31524PT:/opt/nitter$ `

I have no idea why it doesn't work for me.

**Comment by cmj** [comment]
@dhdurgee Thanks. So it looks like i was able to get the original Twitter oauth script working...

Which means you can pretty much scrap this branch and stick with master. Either this one (has a few more features) or the upstream, zedeus guest_accounts branch.

The new-old auth script: https://github.com/cmj/nitter/blob/master/twitter_oauth.sh

Just follow the instructions in README, dump the output from the script to guest_accounts.jsonl and you should be set.

I'll keep this branch and the cookies script around for those who want it, but it's *apparently* not needed... And doesn't support multiple accounts...
